## PO cycle — applying pipeline:fix to #867

CI is red because `TestMatchResourcePattern` fails on Windows-latest:

```
--- FAIL: TestMatchResourcePattern/single_wildcard_does_not_cross_slash (0.00s)
--- FAIL: TestMatchResourcePattern/wildcard_at_root_does_not_cross_slash (0.00s)
features/tenant/security/cross_tenant_access_test.go:257
```

Linux + macOS Native Build: PASS.
Windows Native Build: FAIL.

This is in #867's scope — the `matchResourcePattern` function the story implements. The "single wildcard does not cross slash" semantic is broken on Windows, almost certainly because the implementation treats the path separator as platform-dependent (`filepath.Separator`) when it should be a hard-coded `/` for resource pattern matching. Resource patterns are domain identifiers, not filesystem paths, so they must use `/` regardless of OS.

Fix direction (for the fix agent):
- Check `features/tenant/security/cross_tenant_access.go` — wherever `matchResourcePattern` splits or scans for separators, ensure it uses literal `'/'` (not `filepath.Separator` or `os.PathSeparator`).
- If `path/filepath` is imported, swap to `path` (the OS-agnostic version) or remove the dependency entirely and implement the wildcard match by hand.
- Verify the test cases at `cross_tenant_access_test.go:257` (`single_wildcard_does_not_cross_slash` and `wildcard_at_root_does_not_cross_slash`) capture the intended invariant; if so, the fix is in the implementation, not the test.

The other failure observed in the same job (`features/controller/api` 300s timeout) is unrelated to this PR's scope and likely a flake from native-Windows test slowness — leave that alone, it'll resolve on rerun.

Dispatching fix agent.

--- a/features/tenant/security/cross_tenant_access.go
+++ b/features/tenant/security/cross_tenant_access.go
@@ -5,7 +5,7 @@ package security
 import (
 	"context"
 	"fmt"
-	"path/filepath"
+	"path"
 	"strings"
 	"sync"
 	"time"
@@ -459,14 +459,16 @@ func (ctav *CrossTenantAccessValidator) isTimeInRange(currentTime, timeRange str
 	return curMin >= startMin || curMin <= endMin, nil
 }
 
-// matchResourcePattern reports whether resourceID matches pattern using filepath.Match
+// matchResourcePattern reports whether resourceID matches pattern using path.Match
 // semantics. The wildcard * matches any sequence of non-separator characters and does
-// NOT cross a path separator (/). Returns (false, error) for syntactically invalid patterns.
+// NOT cross a path separator (/). Uses path.Match (not filepath.Match) so that /
+// is always the separator regardless of OS. Returns (false, error) for syntactically
+// invalid patterns.
 func (ctav *CrossTenantAccessValidator) matchResourcePattern(resourceID, pattern string) (bool, error) {
 	if pattern == "" {
 		return false, fmt.Errorf("empty resource pattern")
 	}
-	matched, err := filepath.Match(pattern, resourceID)
+	matched, err := path.Match(pattern, resourceID)
 	if err != nil {
 		return false, fmt.Errorf("invalid resource pattern %q: %w", pattern, err)
 	}

--- a/features/tenant/security/cross_tenant_access_test.go
+++ b/features/tenant/security/cross_tenant_access_test.go
@@ -166,7 +166,7 @@ func TestIsTimeInRange(t *testing.T) {
 	}
 }
 
-// TestMatchResourcePattern covers filepath.Match-based glob pattern evaluation.
+// TestMatchResourcePattern covers path.Match-based glob pattern evaluation.
 func TestMatchResourcePattern(t *testing.T) {
 	v := NewCrossTenantAccessValidator()
 

